cubical/data_handler/TiggerSourceProvider.py

```diff
@@ -150,42 +144,38 @@ def gaussian_lm(self, context):
         for ind, source in enumerate(self._gau_sources[lg:ug]):
 
             ra, dec = source.pos.ra, source.pos.dec
-            lm[ind,0], lm[ind,1] = radec_to_lm(ra, dec, self._phase_center)
+            lm[ind, 0], lm[ind, 1] = radec_to_lm(ra, dec, self._phase_center)
 
         return lm
 
     def gaussian_stokes(self, context):
         """ Return a stokes parameter array to Montblanc """
 
-        # Get the extents of the time, baseline and chan dimension
-        (lt, ut), (lg, ug) = context.dim_extents('ntime', 'ngsrc')
-
+        # Get the extents of the source, time and channel dims
+        (lg, ug), (lt, ut), (lc, uc) = context.dim_extents('ngsrc',
+                                                           'ntime',
+                                                           'nchan')
+        # (npsrc, ntime, nchan, 4)
         stokes = np.empty(context.shape, context.dtype)
 
-        for ind, source in enumerate(self._gau_sources[lg:ug]):
-            stokes[ind,:,0] = source.flux.I
-            stokes[ind,:,1] = source.flux.Q
-            stokes[ind,:,2] = source.flux.U
-            stokes[ind,:,3] = source.flux.V
-
-        return stokes
-
-
-    def gaussian_alpha(self, context):
-        """ Returns a spectral index (alpha) array to Montblanc """
-
-        alpha = np.empty(context.shape, context.dtype)
-
-        (lg, ug) = context.dim_extents('ngsrc')
+        f = self._freqs[None, lc:uc]
 
         for ind, source in enumerate(self._gau_sources[lg:ug]):
             try:
-                alpha[ind] = source.spectrum.spi
-            except:
-                alpha[ind] = 0
+                rf = source.spectrum.freq0
+            except AttributeError:
+                # TODO(jkenyon, osmirnov)
+                # I think this should be "or 1"
+                rf = self._sm.freq0 or 0
+
+            a = source.spectrum.spi
 
-        return alpha
+            stokes[ind, :, :, 0] = source.flux.I[None, None]*(f/rf)**a
```

Comment:
Same here. Even better, have a define_spectrum() method maybe? Since the code is being resued.